src/metabase/util/date.clj

```diff
@@ -13,32 +13,113 @@
            [org.joda.time DateTime DateTimeZone]
            [org.joda.time.format DateTimeFormatter]))
 
+(def ^{:tag     TimeZone
+       :dynamic true
+       :doc     "Timezone to be used when formatting timestamps for display or for the data (pre aggregation)"}
+  *report-timezone*)
+
+(def ^{:dynamic true
+       :doc     "The timezone of the data being queried. Today this is the same as the database timezone."
+       :tag     TimeZone}
+  *data-timezone*)
+
+(defprotocol ITimeZoneCoercible
+  "Coerce object to `java.util.TimeZone`"
+  (coerce-to-timezone ^TimeZone [this]
+    "Coerce `this` to `java.util.TimeZone`"))
+
+(extend-protocol ^:private ITimeZoneCoercible
+  String       (coerce-to-timezone [this]
+                 (TimeZone/getTimeZone this))
+  TimeZone     (coerce-to-timezone [this]
+                 this)
+  DateTimeZone (coerce-to-timezone [this]
+                 (.toTimeZone this)))
+
+(def ^TimeZone utc
+  "UTC TimeZone"
+  (coerce-to-timezone "UTC"))
+
+(def ^:private jvm-timezone
+  (delay (coerce-to-timezone (System/getProperty "user.timezone"))))
+
+(defn- warn-on-timezone-conflict
```

Comment:
Like this a lot, we should really figure out a way to warn admins in-app. 

paging @mazameli @kdoh